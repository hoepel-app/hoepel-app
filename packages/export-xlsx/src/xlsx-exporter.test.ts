import { XlsxExporter } from './xlsx-exporter'
import {
  Address,
  Child,
  ContactPerson,
  Crew,
  Price,
  Shift,
  DayDate,
} from '@hoepel.app/types'

describe('XlsxExporter', () => {
  const exporter = new XlsxExporter()

  const child1 = Child.empty()
    .withFirstName('Kind 1')
    .withLastName('Achter')
    .withId('child-id-1')
    .withBirthDate({ day: 2, month: 3, year: 2014 })
    .withPhoneContact([
      { phoneNumber: '+1111' },
      { phoneNumber: '+2222', comment: 'comment' },
    ])
    .withEmail(['aoeu@test.com', 'test@hello.com'])
    .withGender('male')
    .withUitpasNumber('11-1111-11')
    .withRemarks('Remarks here')
    .withAddress(
      new Address({
        street: 'Straat',
        number: '12A',
        city: 'Stad',
        zipCode: 4444,
      })
    )
  const child2 = Child.empty()
    .withFirstName('Kind 2')
    .withLastName('Achter')
    .withId('child-id-2')
    .withContactPeople([
      { relationship: 'Mama', contactPersonId: 'contact-id-1' },
    ])
  const child3 = Child.empty()
    .withFirstName('Kind 3')
    .withLastName('Achter')
    .withId('child-id-3')

  const crew1 = Crew.empty()
    .withId('crew-id-1')
    .withFirstName('Animator 1')
    .withLastName('Achernaam')
    .withBirthDate({ day: 2, month: 3, year: 2014 })
    .withPhoneContact([
      { phoneNumber: '+2132' },
      { phoneNumber: '+4321', comment: 'comment' },
    ])
    .withEmail(['aoeu@test.com', 'hello@example.com'])
    .withAddress(
      new Address({
        street: 'Straat',
        number: '12A',
        city: 'Stad',
        zipCode: 4444,
      })
    )
    .withActive(false)
    .withBankAccount('BE44 4444 4444')
    .withYearStarted(2019)
    .withCertificates({
      hasPlayworkerCertificate: true,
      hasTeamleaderCertificate: true,
      hasTrainerCertificate: true,
    })
    .withRemarks('Remarks here')
  const crew2 = Crew.empty()
    .withId('crew-id-2')
    .withFirstName('Animator 2')
    .withLastName('Achernaam')
    .withCertificates({
      hasPlayworkerCertificate: true,
      hasTrainerCertificate: false,
      hasTeamleaderCertificate: false,
    })
  const crew3 = Crew.empty()
    .withId('crew-id-3')
    .withFirstName('Animator 3')
    .withLastName('Achernaam')

  const shift1 = new Shift({
    id: 'shift-id-1',
    childrenCanBePresent: true,
    crewCanBePresent: true,
    dayId: '2019-04-03',
    price: new Price({ euro: 5, cents: 50 }),
    kind: 'Voormiddag',
    description: 'Omschrijving 1',
    startAndEnd: {
      start: { hour: 9, minute: 0 },
      end: { hour: 12, minute: 0 },
    },
  })
  const shift2 = new Shift({
    id: 'shift-id-2',
    childrenCanBePresent: true,
    crewCanBePresent: true,
    dayId: '2019-04-03',
    price: new Price({ euro: 2, cents: 50 }),
    kind: 'Namiddag',
    description: 'Omschrijving 2',
    startAndEnd: {
      start: { hour: 13, minute: 0 },
      end: { hour: 17, minute: 30 },
    },
  })
  const shift3 = new Shift({
    id: 'shift-id-4',
    childrenCanBePresent: true,
    crewCanBePresent: true,
    dayId: '2019-08-05',
    price: new Price({ euro: 20, cents: 0 }),
    kind: 'Externe activiteit',
    description: 'Bellewaerde met de tieners',
  })

  const contactPerson1 = ContactPerson.empty()
    .withId('contact-id-1')
    .withFirstName('Mieke')
    .withLastName('Contact')
    .withAddress({
      street: 'Contactstraat',
      number: '123',
      zipCode: 7777,
      city: 'Stadt',
    })
  const contactPerson2 = ContactPerson.empty()
    .withId('contact-id-2')
    .withFirstName('Mieke')
    .withLastName('Contact')

  it('createChildList', () => {
    const res = exporter.createChildList([child1, child2, child3])

    expect(res.filename).toBe('Alle kinderen')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Alle kinderen')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createChildrenWithCommentList', () => {
    const res = exporter.createChildrenWithCommentList([child1])

    expect(res.filename).toBe('Kinderen met opmerking')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Kinderen met opmerking')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createCrewMemberList', () => {
    const res = exporter.createCrewMemberList([crew1, crew2, crew3])

    expect(res.filename).toBe('Alle animatoren')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Alle animatoren')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createCrewMembersAttendanceList', () => {
    const res = exporter.createCrewMembersAttendanceList(
      [crew1, crew2, crew3],
      [shift2, shift1, shift3],
      [
        {
          shiftId: 'shift-id-2',
          attendances: {
            'crew-id-2': { didAttend: true },
            'crew-id-1': { didAttend: true },
            'crew-id-3': { didAttend: false },
          },
        },
        {
          shiftId: 'shift-id-1',
          attendances: { 'crew-id-1': { didAttend: true } },
        },
      ],
      2019
    )

    expect(res.filename).toBe('Aanwezigheden animatoren 2019')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Aanwezigheden animatoren 2019')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createChildAttendanceList', () => {
    const res = exporter.createChildAttendanceList(
      [child1, child2, child3],
      [shift2, shift1, shift3],
      [
        {
          shiftId: 'shift-id-2',
          attendances: {
            'child-id-2': {
              didAttend: true,
              amountPaid: { euro: 5, cents: 0 },
            },
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 6, cents: 0 },
            },
            'child-id-3': {
              didAttend: false,
              amountPaid: { euro: 2, cents: 50 },
            },
          },
        },
        {
          shiftId: 'shift-id-1',
          attendances: {
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 7, cents: 20 },
            },
          },
        },
      ],
      2019
    )

    expect(res.filename).toBe('Aanwezigheden kinderen 2019')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Aanwezigheden kinderen 2019')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createDayOverview', () => {
    const res = exporter.createDayOverview(
      [child1, child2, child3],
      [shift2, shift1, shift3],
      [
        {
          shiftId: 'shift-id-2',
          attendances: {
            'child-id-2': {
              didAttend: true,
              amountPaid: { euro: 5, cents: 0 },
            },
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 6, cents: 0 },
            },
            'child-id-3': {
              didAttend: false,
              amountPaid: { euro: 2, cents: 50 },
              ageGroupName: 'Kleuters',
            },
          },
        },
        {
          shiftId: 'shift-id-1',
          attendances: {
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 7, cents: 20 },
              ageGroupName: 'Tieners',
            },
          },
        },
      ],
      new DayDate({ year: 2019, month: 4, day: 3 })
    )

    expect(res.filename).toBe('Overzicht voor 03-04-2019')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Aanwezigheden kinderen')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createAllFiscalCertificates', () => {
    const res = exporter.createAllFiscalCertificates(
      [child1, child2, child3],
      [contactPerson1, contactPerson2],
      [shift2, shift1, shift3],
      [
        {
          shiftId: 'shift-id-2',
          attendances: {
            'child-id-2': {
              didAttend: true,
              amountPaid: { euro: 5, cents: 0 },
            },
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 6, cents: 0 },
            },
            'child-id-3': {
              didAttend: false,
              amountPaid: { euro: 2, cents: 50 },
            },
          },
        },
        {
          shiftId: 'shift-id-1',
          attendances: {
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 7, cents: 20 },
            },
          },
        },
      ],
      2019
    )

    expect(res.filename).toBe('Data fiscale attesten 2019')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Data fiscale attesten 2019')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })

  it('createChildrenPerDayList', () => {
    const res = exporter.createChildrenPerDayList(
      [shift2, shift1, shift3],
      [
        {
          shiftId: 'shift-id-2',
          attendances: {
            'child-id-2': {
              didAttend: true,
              amountPaid: { euro: 5, cents: 0 },
            },
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 6, cents: 0 },
            },
            'child-id-3': {
              didAttend: false,
              amountPaid: { euro: 2, cents: 50 },
            },
          },
        },
        {
          shiftId: 'shift-id-1',
          attendances: {
            'child-id-1': {
              didAttend: true,
              amountPaid: { euro: 7, cents: 20 },
            },
          },
        },
      ],
      2019
    )

    expect(res.filename).toBe('Aantal unieke kinderen per dag 2019')
    expect(res.worksheets).toHaveLength(1)
    expect(res.worksheets[0].name).toBe('Unieke kinderen per dag 2019')
    expect(res.worksheets[0].name.length).toBeLessThan(31) // Excel worksheet names must be <31 characters

    expect(res).toMatchSnapshot()
  })
})
