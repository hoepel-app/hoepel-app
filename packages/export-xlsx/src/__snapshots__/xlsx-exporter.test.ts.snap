// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XlsxExporter createAllFiscalCertificates 1`] = `
Object {
  "filename": "Data fiscale attesten 2019",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "values": Array [
            "",
            "",
            "",
            "Voornaam",
            "Kind 1",
            "Kind 2",
          ],
          "width": 20,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Familienaam",
            "Achter",
            "Achter",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Totaal (incl. korting)",
            Price {
              "cents": 20,
              "euro": 13,
            },
            Price {
              "cents": 0,
              "euro": 5,
            },
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Geboortedatum",
            DayDate {
              "day": 2,
              "month": 3,
              "year": 2014,
            },
            undefined,
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Contactpersoon",
            "",
            "Mieke Contact",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Straat en nummer",
            "Straat 12A",
            "Contactstraat 123",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Postcode",
            4444,
            7777,
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "",
            "",
            "",
            "Stad",
            "Stad",
            "Stadt",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            "Dag",
            "Type",
            "Prijs",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Voormiddag",
            Price {
              "cents": 50,
              "euro": 5,
            },
            "Omschrijving 1",
            true,
            false,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Namiddag",
            Price {
              "cents": 50,
              "euro": 2,
            },
            "Omschrijving 2",
            true,
            true,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 5,
              "month": 8,
              "year": 2019,
            },
            "Externe activiteit",
            Price {
              "cents": 0,
              "euro": 20,
            },
            "Bellewaerde met de tieners",
            false,
            false,
          ],
          "width": 22,
        },
      ],
      "name": "Data fiscale attesten 2019",
    },
  ],
}
`;

exports[`XlsxExporter createChildAttendanceList 1`] = `
Object {
  "filename": "Aanwezigheden kinderen 2019",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "values": Array [
            "",
            "",
            "Voornaam",
            "Kind 1",
            "Kind 2",
          ],
          "width": 20,
        },
        Object {
          "values": Array [
            "",
            "",
            "Familienaam",
            "Achter",
            "Achter",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Voormiddag",
            "Omschrijving 1",
            true,
            false,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Namiddag",
            "Omschrijving 2",
            true,
            true,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 5,
              "month": 8,
              "year": 2019,
            },
            "Externe activiteit",
            "Bellewaerde met de tieners",
            false,
            false,
          ],
          "width": 22,
        },
      ],
      "name": "Aanwezigheden kinderen 2019",
    },
  ],
}
`;

exports[`XlsxExporter createChildList 1`] = `
Object {
  "filename": "Alle kinderen",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "hideColumn": true,
          "title": "Id",
          "values": Array [
            "child-id-1",
            "child-id-2",
            "child-id-3",
          ],
          "width": 20,
        },
        Object {
          "title": "Voornaam",
          "values": Array [
            "Kind 1",
            "Kind 2",
            "Kind 3",
          ],
          "width": 20,
        },
        Object {
          "title": "Familienaam",
          "values": Array [
            "Achter",
            "Achter",
            "Achter",
          ],
          "width": 25,
        },
        Object {
          "title": "Geboortedatum",
          "values": Array [
            DayDate {
              "day": 2,
              "month": 3,
              "year": 2014,
            },
            undefined,
            undefined,
          ],
          "width": 15,
        },
        Object {
          "title": "Telefoonnummer",
          "values": Array [
            "+1111, +2222 (comment)",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Geregistreerd door",
          "values": Array [
            "",
            "",
            "My display name",
          ],
          "width": 25,
        },
        Object {
          "title": "Geregistreerd door (emailadres)",
          "values": Array [
            "",
            "test@email.com",
            "my@email.com",
          ],
          "width": 25,
        },
        Object {
          "title": "Emailadres",
          "values": Array [
            "aoeu@test.com, test@hello.com",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Adres",
          "values": Array [
            "Straat 12A, 4444 Stad",
            "",
            "",
          ],
          "width": 30,
        },
        Object {
          "title": "Gender",
          "values": Array [
            "Man",
            "",
            "",
          ],
        },
        Object {
          "title": "Uitpasnummer",
          "values": Array [
            "11-1111-11",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Opmerkingen",
          "values": Array [
            "Remarks here",
            "",
            "",
          ],
          "width": 75,
        },
      ],
      "name": "Alle kinderen",
    },
  ],
}
`;

exports[`XlsxExporter createChildrenPerDayList 1`] = `
Object {
  "filename": "Aantal unieke kinderen per dag 2019",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "title": "Dag",
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            DayDate {
              "day": 5,
              "month": 8,
              "year": 2019,
            },
          ],
          "width": 20,
        },
        Object {
          "title": "Aantal unieke kinderen",
          "values": Array [
            2,
            0,
          ],
          "width": 25,
        },
      ],
      "name": "Unieke kinderen per dag 2019",
    },
  ],
}
`;

exports[`XlsxExporter createChildrenWithCommentList 1`] = `
Object {
  "filename": "Kinderen met opmerking",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "hideColumn": true,
          "title": "Id",
          "values": Array [
            "child-id-1",
          ],
          "width": 20,
        },
        Object {
          "title": "Voornaam",
          "values": Array [
            "Kind 1",
          ],
          "width": 20,
        },
        Object {
          "title": "Familienaam",
          "values": Array [
            "Achter",
          ],
          "width": 25,
        },
        Object {
          "title": "Geboortedatum",
          "values": Array [
            DayDate {
              "day": 2,
              "month": 3,
              "year": 2014,
            },
          ],
          "width": 15,
        },
        Object {
          "title": "Telefoonnummer",
          "values": Array [
            "+1111, +2222 (comment)",
          ],
          "width": 25,
        },
        Object {
          "title": "Geregistreerd door",
          "values": Array [
            "Name",
          ],
          "width": 25,
        },
        Object {
          "title": "Geregistreerd door (emailadres)",
          "values": Array [
            "test@mail.com",
          ],
          "width": 25,
        },
        Object {
          "title": "Emailadres",
          "values": Array [
            "aoeu@test.com, test@hello.com",
          ],
          "width": 25,
        },
        Object {
          "title": "Adres",
          "values": Array [
            "Straat 12A, 4444 Stad",
          ],
          "width": 30,
        },
        Object {
          "title": "Gender",
          "values": Array [
            "Man",
          ],
        },
        Object {
          "title": "Uitpasnummer",
          "values": Array [
            "11-1111-11",
          ],
          "width": 25,
        },
        Object {
          "title": "Opmerkingen",
          "values": Array [
            "Remarks here",
          ],
          "width": 75,
        },
      ],
      "name": "Kinderen met opmerking",
    },
  ],
}
`;

exports[`XlsxExporter createCrewMemberList 1`] = `
Object {
  "filename": "Alle animatoren",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "title": "Voornaam",
          "values": Array [
            "Animator 1",
            "Animator 2",
            "Animator 3",
          ],
          "width": 20,
        },
        Object {
          "title": "Familienaam",
          "values": Array [
            "Achernaam",
            "Achernaam",
            "Achernaam",
          ],
          "width": 25,
        },
        Object {
          "title": "Geboortedatum",
          "values": Array [
            DayDate {
              "day": 2,
              "month": 3,
              "year": 2014,
            },
            undefined,
            undefined,
          ],
          "width": 15,
        },
        Object {
          "title": "Telefoonnummer",
          "values": Array [
            "+2132, +4321 (comment)",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Emailadres",
          "values": Array [
            "aoeu@test.com, hello@example.com",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Adres",
          "values": Array [
            "Straat 12A, 4444 Stad",
            "",
            "",
          ],
          "width": 30,
        },
        Object {
          "title": "Actief",
          "values": Array [
            "Nee",
            "Ja",
            "Ja",
          ],
        },
        Object {
          "title": "Rekeningnummer",
          "values": Array [
            "BE44 4444 4444",
            "",
            "",
          ],
          "width": 25,
        },
        Object {
          "title": "Gestart in",
          "values": Array [
            2019,
            undefined,
            undefined,
          ],
        },
        Object {
          "title": "Attesten",
          "values": Array [
            "Attest animator, Attest hoofdanimator, Attest instructeur",
            "Attest animator",
            "",
          ],
          "width": 35,
        },
        Object {
          "title": "Opmerkingen",
          "values": Array [
            "Remarks here",
            "",
            "",
          ],
          "width": 75,
        },
      ],
      "name": "Alle animatoren",
    },
  ],
}
`;

exports[`XlsxExporter createCrewMembersAttendanceList 1`] = `
Object {
  "filename": "Aanwezigheden animatoren 2019",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "values": Array [
            "",
            "",
            "Voornaam",
            "Animator 1",
            "Animator 2",
          ],
          "width": 20,
        },
        Object {
          "values": Array [
            "",
            "",
            "Familienaam",
            "Achernaam",
            "Achernaam",
          ],
          "width": 25,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Voormiddag",
            "Omschrijving 1",
            true,
            false,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 3,
              "month": 4,
              "year": 2019,
            },
            "Namiddag",
            "Omschrijving 2",
            true,
            true,
          ],
          "width": 22,
        },
        Object {
          "values": Array [
            DayDate {
              "day": 5,
              "month": 8,
              "year": 2019,
            },
            "Externe activiteit",
            "Bellewaerde met de tieners",
            false,
            false,
          ],
          "width": 22,
        },
      ],
      "name": "Aanwezigheden animatoren 2019",
    },
  ],
}
`;

exports[`XlsxExporter createDayOverview 1`] = `
Object {
  "filename": "Overzicht voor 03-04-2019",
  "worksheets": Array [
    Object {
      "columns": Array [
        Object {
          "title": "Voornaam",
          "values": Array [
            "Kind 1",
            "Kind 2",
            "Kind 1",
          ],
          "width": 20,
        },
        Object {
          "title": "Achternaam",
          "values": Array [
            "Achter",
            "Achter",
            "Achter",
          ],
          "width": 25,
        },
        Object {
          "hideIfNoSetValues": true,
          "title": "Leeftijdsgroep",
          "values": Array [
            "Tieners",
            undefined,
            undefined,
          ],
          "width": 20,
        },
        Object {
          "title": "Soort",
          "values": Array [
            "Voormiddag",
            "Namiddag",
            "Namiddag",
          ],
          "width": 20,
        },
        Object {
          "hideIfNoSetValues": true,
          "title": "Beschrijving",
          "values": Array [
            "Omschrijving 1",
            "Omschrijving 2",
            "Omschrijving 2",
          ],
          "width": 25,
        },
        Object {
          "hideIfNoSetValues": true,
          "title": "Locatie",
          "values": Array [
            "",
            "Gewoon op het speelplein",
            "Gewoon op het speelplein",
          ],
          "width": 25,
        },
        Object {
          "hideIfNoSetValues": true,
          "title": "Bubbel",
          "values": Array [
            undefined,
            undefined,
            "Bubbel Rood",
          ],
          "width": 25,
        },
      ],
      "name": "Aanwezigheden kinderen",
    },
  ],
}
`;
